pkg/validation/validation.go

```diff
@@ -112,6 +119,26 @@ func NodePorts(contour *operatorv1alpha1.Contour) error {
 	return nil
 }
 
+// loadBalancer validates LoadBalancer "address" parameter of contour, returning an
+// error if "address" does not meet API specification.
+func loadBalancerAddress(contour *operatorv1alpha1.Contour) error {
+	if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type == operatorv1alpha1.AzureLoadBalancerProvider &&
+		contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Azure.Address != nil {
+		validationIP := net.ParseIP(*contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Azure.Address)
+		if validationIP == nil {
+			return fmt.Errorf("wrong LoadBalancer address format, should be string with IPv4 or IPv6 format")
+		}
+	} else if contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.Type == operatorv1alpha1.GCPLoadBalancerProvider &&
```

Comment:
You should also do a nil check for `contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.GCP`.